pallets/parachain-staking/src/lib.rs

```diff
@@ -170,6 +173,13 @@ pub mod pallet {
 		/// Handler to distribute a collator's reward.
 		/// To use the default implementation of minting rewards, specify the type `()`.
 		type PayoutCollatorReward: PayoutCollatorReward<Self>;
+		/// Handler to notify the runtime when a collator is inactive.
+		/// The default behavior is to mark the collator as offline.
+		/// If you need to use the default implementation, specify the type `()`.
+		type OnInactiveCollator: OnInactiveCollator<Self>;
+		// Killswitch to enable/disable marking offline feature.
+		#[pallet::constant]
+		type EnableMarkingOffline: Get<bool>;
```

Comment:
I don't think there's much value in having a killswitch implemented as a runtime-constant. It basically means that a runtime upgrade would be required to toggle the switch -- if we had to do a code change and runtime upgrade anyway, we could just remove the feature.

I suggest implementing it as a pallet storage item, similar to what's done elsewhere.

Reply:
Yeah I was hesitating to implement the killswitch as storage as well. I decided to implement it as a constant to avoid adding an extra storage item (that could result more expensive). But I think the improvement is not really worth it, so I will implement it as a storage item.

Reply:
It's also a pattern we have repeated many times, maybe we need something like the Precompile Registry to abstract it...